pkg/controllers/infracluster/infracluster_controller.go

```diff
@@ -0,0 +1,265 @@
+package infracluster
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strconv"
+
+	"github.com/go-logr/logr"
+
+	cerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"k8s.io/client-go/rest"
+	awsv1 "sigs.k8s.io/cluster-api-provider-aws/v2/api/v1beta2"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/cluster-capi-operator/pkg/controllers"
+	"github.com/openshift/cluster-capi-operator/pkg/operatorstatus"
+)
+
+const (
+	// Controller conditions for the Cluster Operator resource
+	InfraClusterControllerAvailableCondition = "InfraClusterControllerAvailable"
+	InfraClusterControllerDegradedCondition  = "InfraClusterControllerDegraded"
+
+	defaultCAPINamespace    = "openshift-cluster-api"
+	defaultInfraClusterName = "cluster"
+	clusterOperatorName     = "cluster-api"
+	// This is the managedByAnnotation value that this controller sets by default when it creates an InfraCluster object.
+	// If the managedByAnnotation key is set, and it has this as the value, it means this controller is managing the InfraCluster.
+	managedByAnnotationValueClusterCAPIOperator = ""
```

Comment:
Can we change this to something before we ship CAPI as GA?

Reply:
ok, I think I can change it to `cluster-capi-operator-infracluster-controller`